qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -195,6 +203,172 @@ def main(self):
+        stop_node(self.nodes[0], 0)
+        args = ["-debug",
+                "-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK"] + \
+               self.extra_args
+        self.nodes[0] = start_node(0, self.options.tmpdir,
+                                   args)
+        self.create_masternodes()
+        # create connected simple nodes
+        for i in range(0, self.num_nodes - self.mn_count - 1):
+            self.create_simple_node()
+        set_mocktime(get_mocktime() + 1)
+        set_node_times(self.nodes, get_mocktime())
+        self.nodes[0].generate(1)
+        # sync nodes
+        self.sync_all()
+        set_mocktime(get_mocktime() + 1)
+        set_node_times(self.nodes, get_mocktime())
+        sync_masternodes(self.nodes)
+        for i in range(1, self.mn_count + 1):
+            res = self.nodes[0].masternode("start-alias", "mn%d" % i)
+            assert (res["result"] == 'successful')
+        sync_masternodes(self.nodes)
+        mn_info = self.nodes[0].masternodelist("status")
+        assert (len(mn_info) == self.mn_count)
+        for status in mn_info.values():
+            assert (status == 'ENABLED')
+
+    def enforce_masternode_payments(self):
+        self.nodes[0].spork('SPORK_8_MASTERNODE_PAYMENT_ENFORCEMENT', 0)
+
+    def create_raw_trx(self, node_from, node_to, amount, min_inputs, max_inputs):
+        assert (min_inputs <= max_inputs)
+        # fill inputs
+        inputs = []
+        balances = node_from.listunspent()
+        in_amount = 0.0
+        last_amount = 0.0
+        for tx in balances:
+            if len(inputs) < min_inputs:
+                input = {}
+                input["txid"] = tx['txid']
+                input['vout'] = tx['vout']
+                in_amount += float(tx['amount'])
+                inputs.append(input)
+            elif in_amount > amount:
+                break
+            elif len(inputs) < max_inputs:
+                input = {}
+                input["txid"] = tx['txid']
+                input['vout'] = tx['vout']
+                in_amount += float(tx['amount'])
+                inputs.append(input)
+            else:
+                input = {}
+                input["txid"] = tx['txid']
+                input['vout'] = tx['vout']
+                in_amount -= last_amount
+                in_amount += float(tx['amount'])
+                inputs[-1] = input
+            last_amount = float(tx['amount'])
+
+        assert (len(inputs) > 0)
+        assert (in_amount > amount)
+        # fill outputs
+        receiver_address = node_to.getnewaddress()
+        change_address = node_from.getnewaddress()
+        fee = 0.001
+        outputs = {}
+        outputs[receiver_address] = satoshi_round(amount)
+        outputs[change_address] = satoshi_round(in_amount - amount - fee)
+        rawtx = node_from.createrawtransaction(inputs, outputs)
+        return node_from.signrawtransaction(rawtx)
+
+    def check_IX_lock(self, txid, node):
```

Comment:
Maybe the name here is sub-optimal? It's not creating a lock but waiting for one

Reply:
It says `check`, not `create`, so I guess it's ok-ish but, I agree, `wait` would be clearer probably. Also let's get rid of `IX` ;) Smth like `wait_for_ instantlock` maybe?

Reply:
Ok,  I'll rename it to `wait_for_instantlock`

Reply:
Ah sorry, I somehow misread the name and thus was confused.